chall02/ksappi.c

```diff
@@ -0,0 +1,82 @@
+#include <string.h>
+#include <stdlib.h>
+
+char	*ft_strnew(size_t size)
+{
+	char	*area;
+	size_t	i;
+
+	area = (char *)malloc(sizeof(char) * (size + 1));
+	if (!area)
+		return (NULL);
+	i = 0;
+	while (i < size + 1)
+	{
+		area[i] = '\0';
+		++i;
+	}
+	return (area);
+}
+
+char	*ft_itoa_base(unsigned long long nb, char base, char capitalise)
+{
+	char				*str;
+	char				digits[16];
+	int					len;
+	unsigned long long	nb_copy;
+
+	nb_copy = nb;
+	len = 0;
+	while ((nb_copy /= base) > 0)
+		++len;
+	if (base < 2 || base > 16 || !(str = ft_strnew(len + 1)))
+		return (NULL);
+	if (capitalise)
+		memcpy(digits, "0123456789ABCDEF", 16 * sizeof(char));
+	else
+		memcpy(digits, "0123456789abcdef", 16 * sizeof(char));
+	while (len > -1)
+	{
+		str[len] = digits[nb % base];
+		nb /= base;
+		--len;
+	}
+	return (str);
+}
+int check_if_valid_value(int color)
+{
+  if (color < 0 || color > 255)
+    return 0;
+  else
+    return 1;
```

Comment:
could have been one-lined, but I appreciate you checked the input given by the user.